core/src/Kokkos_Complex.hpp

```diff
@@ -705,23 +716,97 @@ KOKKOS_INLINE_FUNCTION RealType abs(const complex<RealType>& x) {
 }
 
 //! Power of a complex number
-template <class RealType>
-KOKKOS_INLINE_FUNCTION Kokkos::complex<RealType> pow(const complex<RealType>& x,
-                                                     const RealType& e) {
-  RealType r = abs(x);
-#ifdef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_SYCL
-  using sycl::atan;
-  using sycl::cos;
-  using sycl::pow;
-  using sycl::sin;
-#else
-  using std::atan;
-  using std::cos;
-  using std::pow;
-  using std::sin;
-#endif
-  RealType phi = atan(x.imag() / x.real());
-  return pow(r, e) * Kokkos::complex<RealType>(cos(phi * e), sin(phi * e));
+template <class T>
+KOKKOS_INLINE_FUNCTION complex<T> pow(const complex<T>& x, const T& y) {
+  using Kokkos::Experimental::exp;
+  using Kokkos::Experimental::log;
```

Comment:
@dalg24 Why do you change from using atan2 to this algorithm?

Reply:
No good reason.  I changed it back.